go/build.sh

```diff
@@ -9,4 +9,5 @@ popd || exit
 # Build sample app
 
 cd ../opentelemetry-lambda/go/sample-apps/function || exit
+go mod tidy
```

Comment:
This should not be necessary.  The repo shouldn't ever be left in a state where the `go.mod` file is inconsistent.

Reply:
i agree, created the [PR](https://github.com/open-telemetry/opentelemetry-lambda/pull/1644), i will remove this line once the upstream pr is merged.